File: python_instantiator/DatasetJob/tpc_h/TPCHDatasetOperatorManager.py
from python_instantiator.DatasetJob.AbstractDatasetOperatorManager import AbstractDatasetOperatorManager
from python_instantiator.DatasetJob.tpc_h.RegionOperatorManager import RegionOperatorManager
from python_instantiator.DatasetJob.tpc_h.NationOperatorManager import NationOperatorManager
from python_instantiator.DatasetJob.tpc_h.PartOperatorManager import PartOperatorManager
from python_instantiator.DatasetJob.tpc_h.SupplierOperatorManager import SupplierOperatorManager
from python_instantiator.DatasetJob.tpc_h.PartsuppOperatorManager import PartsuppOperatorManager
from python_instantiator.DatasetJob.tpc_h.OrdersOperatorManager import OrdersOperatorManager
from python_instantiator.DatasetJob.tpc_h.CustomerOperatorManager import CustomerOperatorManager
from python_instantiator.DatasetJob.tpc_h.LineitemOperatorManager import LineitemOperatorManager

class TPCHDatasetOperatorManager(AbstractDatasetOperatorManager):
    tables = ["lineitem", "orders", "customer", "partsupp", "part", "supplier", "nation"]
    tables_to_class = {
        "lineitem": LineitemOperatorManager,
        "nation": NationOperatorManager,
        "part": PartOperatorManager,
        "supplier": SupplierOperatorManager,
        "customer": CustomerOperatorManager,
        "orders": OrdersOperatorManager,
        "partsupp": PartsuppOperatorManager
    }
    
    def __init__(self):
        super().__init__() # Don't know if there are no arguments
        #self.start_routine()
        
    def get_table_operator_manager(self, s):
        return self.tables_to_class[s]()
    
    def start_routine(self):
        for i in range(100):
            try:
                table_sequence, join_sequence = self.get_source_table_sequence(5,i)
                for join in join_sequence:
                    print(f"> {join}\n")
                    print(f"Table sequence: {table_sequence}")
            except:
                print("Exception thrown")
            print("-------------------------------")
    
# here is a main function which I will first use as class method
        

File: instantiator/src/main/scala/Generator/DatasetJob/tpc_h/spark_table/TPCHDatasetOperatorManager.scala
package Generator.DatasetJob.tpc_h.spark_table

import Generator.DatasetJob.{AbstractDatasetOperatorManager, AbstractTableOperatorManager, AbstractTableOperatorManagerSpark}

import scala.collection.mutable
import scala.util.Random

/**
  * Created by researchuser7 on 2020-04-20.
  */


object TPCHDatasetOperatorManager extends AbstractDatasetOperatorManager {

  val tables: Seq[String] = Seq(
    "lineitem",
    "orders",
    "customer",
    "partsupp",
    "part",
    "supplier",
    "nation"
  )

  override def getTableOperatorManager(s: String): AbstractTableOperatorManager = s match {
    case "lineitem" => new LineitemOperatorManager() with AbstractTableOperatorManagerSpark
    case "orders" => new  OrdersOperatorManager() with AbstractTableOperatorManagerSpark
    case "customer" => new CustomerOperatorManager() with AbstractTableOperatorManagerSpark
    case "partsupp" => new PartsuppOperatorManager() with AbstractTableOperatorManagerSpark
    case "part" => new PartOperatorManager() with AbstractTableOperatorManagerSpark
    case "supplier" => new SupplierOperatorManager() with AbstractTableOperatorManagerSpark
    case "nation" => new NationOperatorManager() with AbstractTableOperatorManagerSpark
    case _ => throw new Exception(s"Can not find table operator manager for table '$s' ")
  }

  def main(args: Array[String]): Unit = {
    for (i <- 0 to 100){
      try{
        val (tableSequence, joinSequence) = getSourceTableSequence(5, i)
        joinSequence.foreach( j => println(s"> $j"))
        println()
        println(s"Table sequence: $tableSequence")
      } catch {
        case ex:Exception => println(ex)
      } finally {
        println("---------------------------------------------")
      }
    }
  }
}
